app/assets/javascripts/select-kit/addon/components/category-selector.js

```diff
@@ -27,27 +26,70 @@ export default MultiSelectComponent.extend({
     if (!this.blockedCategories) {
       this.set("blockedCategories", []);
     }
-  },
 
-  content: computed("categories.[]", "blockedCategories.[]", function () {
-    const blockedCategories = makeArray(this.blockedCategories);
-    return Category.list().filter((category) => {
-      if (category.isUncategorizedCategory) {
-        if (this.options?.allowUncategorized !== undefined) {
-          return this.options.allowUncategorized;
+    if (this.siteSettings.lazy_load_categories) {
+      if (this.categoryIds) {
+        if (!Category.hasAsyncFoundAll(this.categoryIds)) {
+          Category.asyncFindByIds(this.categoryIds).then(() => {
```

Comment:
Just curious, what if `categoryIds` is really large? Do we have some sort of limit here?

Reply:
There is no limit yet. At this moment the focus is to make it work and then make it faster.